### PR TITLE
QE NAS-103329 Adding automation testing to create, edit and delete a zvol with API v2

### PR DESCRIPTION
**Pull request by ericbsd**

### CODE REVIEW

File: tests/api2/pool_dataset.py [hunk 1]
```diff
@@ -143,3 +145,72 @@ def test_13_delete_dataset():
...
+        'type': 'VOLUME',
+        'volsize': 163840,
+        'volblocksize': '16K'
+    }
+    results = POST(f"/pool/dataset/", payload)
+    assert results.status_code == 200, results.text
+
+
+@pytest.mark.parametrize('key', ['name', 'type', 'volsize', 'volblocksize'])
+def test_15_verify_output_(key):
```
Comment (sonicaj) [inline tests/api2/pool_dataset.py]: Is the last underscore intentional ?

File: tests/api2/pool_dataset.py [hunk 2]
```diff
@@ -143,3 +145,72 @@ def test_13_delete_dataset():
...
+    payload = {
+        'volsize': 163840,
+        'comments': 'testing zvol'
+    }
+    result = PUT(f'/pool/dataset/id/{zvol_url}/', payload)
+    assert result.status_code == 200, result.text
+
+
+@pytest.mark.parametrize('key', ['volsize'])
+def test_19_verify_update_zvol_output_(key):
```
Comment (sonicaj) [inline tests/api2/pool_dataset.py]: Same here

File: tests/api2/pool_dataset.py [hunk 3]
```diff
@@ -143,3 +145,72 @@ def test_13_delete_dataset():
...
+    assert isinstance(results.json(), dict)
+
+
+@pytest.mark.parametrize('key', ['comments', 'volsize'])
+def test_21_verify_the_query_change_zvol_output_(key):
+    assert results.json()[key]['parsed'] == payload[key], results.text
+
+
+def test_22_delete_zvol():
+    result = DELETE(f'/pool/dataset/id/{zvol_url}/')
```
Comment (sonicaj) [inline tests/api2/pool_dataset.py]: We should verify that indeed no longer exists

File: tests/api2/pool_dataset.py (hunk 1 above)
Comment (ericbsd) [reply tests/api2/pool_dataset.py]: Yes the out put look like that `api2/pool_dataset.py::test_15_verify_output_[name] PASSED`

Comment (ericbsd) [reply tests/api2/pool_dataset.py]: I can remove that no problem

Comment (sonicaj) [reply tests/api2/pool_dataset.py]: I have no objections to this, was wondering if it was a typo. Thank you for elaborating

**Pull request merged**